File: src/Neuralm.Services/Neuralm.Services.TrainingRoomService/Neuralm.Services.TrainingRoomService.Messages/GetOrganismsResponse.cs
using Neuralm.Services.Common.Messages.Abstractions;
using Neuralm.Services.TrainingRoomService.Messages.Dtos;
using System;
using System.Collections.Generic;

namespace Neuralm.Services.TrainingRoomService.Messages
{
    /// <summary>
    /// Represents the <see cref="GetOrganismsResponse"/> class.
    /// </summary>
    public class GetOrganismsResponse : Response
    {
        /// <summary>
        /// Gets the collection of organisms.
        /// </summary>
        public List<OrganismDto> Organisms { get; set; }

        /// <summary>
        /// Initializes an instance of the <see cref="GetOrganismsResponse"/> class.
        /// </summary>
        /// <param name="requestId">The request id.</param>
        /// <param name="organisms">The organisms.</param>
        /// <param name="message">The message.</param>
        /// <param name="success">The success flag.</param>
        public GetOrganismsResponse(Guid requestId, List<OrganismDto> organisms, string message = "", bool success = false) : base(requestId, message, success)
        {
            Organisms = organisms ?? new List<OrganismDto>();
        }
        
        /// <summary>
        /// Initializes an instance of the <see cref="GetOrganismsResponse"/> class.
        /// SERIALIZATION CONSTRUCTOR.
        /// </summary>
        public GetOrganismsResponse()
        {
            
        }
    }
}

File: src/Neuralm.Presentation.Web/src/messages/responses/GetOrganismsResponse.ts
import Response from './Response';
import Organism from '../../models/Organism';

/**
 * Represents the get organisms response class.
 */
export default class GetOrganismsResponse extends Response {
  public organisms: Organism[];

  /**
   * Initializes a new instance of the GetOrganismsResponse class.
   */
  constructor(id: string, requestId: string, dateTime: Date, message: string, success: boolean, organisms: Organism[]) {
    super(id, requestId, dateTime, message, success);
    this.organisms = organisms;
  }
}
